src/libtoast/atm/toast_atm.cpp

```diff
@@ -183,184 +212,199 @@ void toast::atm::sim::simulate( bool save_covmat ) {
-        std::cerr << std::endl;
-        std::cerr << "Realization constructed in " << t2-t1 << " s." << std::endl;
+      std::cerr << std::endl;
+      std::cerr << "Realization constructed in " << t2-t1 << " s." << std::endl;
     }
 
-    return;
+  } catch ( const std::exception& e ) {
+    std::cerr << "WARNING: atm::simulate failed with: " << e.what() << std::endl;
+  }
+
+  return;
 }
 
 
 void toast::atm::sim::get_slice( long &ind_start, long &ind_stop ) {
 
-    // Identify a manageable slice of compressed indices to simulate next
-
-    ind_start = ind_stop;
+  // Identify a manageable slice of compressed indices to simulate next
 
-    long ix_start = full_index[ind_start] / xstride;
-    long ix1 = ix_start;
-    long ix2;
+  ind_start = ind_stop;
 
-    while ( true ) {
-        ix2 = ix1;
-        while ( ix1 == ix2 ) {
-            ++ind_stop;
-            if ( ind_stop == nelem ) break;
-            ix2 = full_index[ind_stop] / xstride;
-        }
-        if ( ind_stop == nelem ) break;
-        if ( ind_stop - ind_start > nelem_sim_max ) break;
-        ix1 = ix2;
-    }
+  long ix_start = full_index[ind_start] / xstride;
+  long ix1 = ix_start;
+  long ix2;
 
-    if ( rank == 0 && verbosity > 0 ) {
-        std::cerr << "X-slice: " << ix_start*xstep << " -- " << ix2*xstep
-        << " ( " << nx*xstep << " )"
-        << "( " << ind_start << " -- "<< ind_stop << " ) "
-        << "( " << nelem << " )"<< std::endl;
+  while ( true ) {
+    ix2 = ix1;
+    while ( ix1 == ix2 ) {
+      ++ind_stop;
+      if ( ind_stop == nelem ) break;
+      ix2 = full_index[ind_stop] / xstride;
     }
-
-    return;
+    if ( ind_stop == nelem ) break;
+    if ( ind_stop - ind_start > nelem_sim_max ) break;
+    ix1 = ix2;
+  }
+
+  if ( rank == 0 && verbosity > 0 ) {
+    std::cerr << "X-slice: " << ix_start*xstep << " -- " << ix2*xstep
+	      << " ( " << nx*xstep << " )"
+	      << "( " << ind_start << " -- "<< ind_stop << " ) "
+	      << "( " << nelem << " )"<< std::endl;
+  }
+
+  return;
 }
 
 
 void toast::atm::sim::smooth() {
 
-    // Replace each vertex with a mean of its immediate vicinity
+  // Replace each vertex with a mean of its immediate vicinity
 
-    double t1 = MPI_Wtime();
+  double t1 = MPI_Wtime();
 
-    double coord[3];
+  double coord[3];
 
-    std::vector<double> smoothed_realization(realization.size());
-    std::vector<double> smoothed_realization_near(realization.size());
-    std::vector<double> smoothed_realization_verynear(realization.size());
+  std::vector<double> smoothed_realization(realization.size());
+  std::vector<double> smoothed_realization_near(realization.size());
+  std::vector<double> smoothed_realization_verynear(realization.size());
 
-    for ( size_t i=0; i < full_index.size(); ++i ) {
-        ind2coord( i, coord );
-        long ix = coord[0] * xstepinv;
-        long iy = coord[1] * ystepinv;
-        long iz = coord[2] * zstepinv;
+  for ( size_t i=0; i < full_index.size(); ++i ) {
```

Comment:
There seem to be several places where the indent got changed from 4 spaces to 2 (or maybe a tab?).  It might just be github's rendering though.